src/Http/Http/src/Internal/RequestCookieCollection.cs

```diff
@@ -117,7 +117,7 @@ public bool ContainsKey(string key)
         return Store.ContainsKey(key);
     }
 
-    public bool TryGetValue(string key, [MaybeNullWhen(false)] out string? value)
+    public bool TryGetValue(string key, [MaybeNullWhen(false)] out string value)
```

Comment:
```suggestion
    public bool TryGetValue(string key, [NotNullWhen(true)] out string? value)
```

> Look at for example TryGetValue from Dictionary

I think dictionary is using `MaybeNullWhen` because `TValue` is generic and `TValue?` would lead to `TryGetValue` producing a nullable which is undesirable. Eg. `Dictionary<MyKey, MyStruct>.TryGetValue(myKey, out Nullable<MyStruct> myStruct)`. For non-generic parameters, I think it's more common to use `NotNullWhen`. [Example](https://github.com/dotnet/runtime/blob/eef6c7f6cb7f73163313d349fbb796f37b4c3996/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs#L381).

Reply:
Yeah, that makes sense to me. Also makes it so the API doesn't change.